clang/lib/Driver/Driver.cpp

```diff
@@ -4934,8 +4934,7 @@ class OffloadingActionBuilder final {
         // with -fsycl-device-only.
         bool IsPreprocessOnly =
             Args.getLastArg(options::OPT_E) ||
-            Args.getLastArg(options::OPT__SLASH_EP, options::OPT__SLASH_P) ||
-            Args.getLastArg(options::OPT_M, options::OPT_MM);
+            Args.getLastArg(options::OPT__SLASH_EP, options::OPT__SLASH_P);
```

Comment:
My initial inklings had me remove these - but now that I've done a bit more digging, keeping this around makes sense due to the interactions with `-fsycl-device-only`.

Reply:
>> keeping this around makes sense due to the interactions with -fsycl-device-only

Can you please elaborate?

Since you have removed the check for `options::OPT_M, options::OPT_MM` for `PreprocessOnly` phase,
I was wondering if we still needed this check in `ConstructPhaseAction` for `case phases::Preprocess:`